includes/class-wc-subscriptions-order.php

```diff
@@ -932,25 +933,66 @@ public static function add_subscriptions_to_view_order_templates( $order_id ) {
 	/**
 	 * Loads the related orders table on the view subscription page
 	 *
-	 * @since 1.0.0 - Migrated from WooCommerce Subscriptions v2.0
+	 * @since 1.0.0 Migrated from WooCommerce Subscriptions v2.0.
+	 *
+	 * @param WC_Subscription $subscription The subscription whose related orders we are interested in.
 	 */
 	public static function get_related_orders_template( $subscription ) {
+		// phpcs:disable WordPress.Security.NonceVerification.Recommended
+		// phpcs:disable WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+		// phpcs:disable WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+		$page = (int) ( $_GET['related_orders_page'] ?? 1 );
+		// phpcs:enable WordPress.Security.NonceVerification.Recommended
+		// phpcs:enable WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+		// phpcs:enable WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
 
-		$subscription_orders = $subscription->get_related_orders();
+		$related_orders      = $subscription->get_paginated_related_orders( 'ids', array( 'parent', 'renewal', 'switch' ), $page, 10 );
+		$subscription_orders = $related_orders->orders;
 
 		if ( 0 !== count( $subscription_orders ) ) {
 			wc_get_template(
 				'myaccount/related-orders.php',
 				array(
 					'subscription_orders' => $subscription_orders,
 					'subscription'        => $subscription,
+					'page'                => $page,
+					'max_num_pages'       => $related_orders->max_num_pages,
 				),
 				'',
 				WC_Subscriptions_Core_Plugin::instance()->get_subscriptions_core_directory( 'templates/' )
 			);
 		}
 	}
 
+	/**
+	 * Introduced to support pagination of the related orders list (within the My Account > View Subscription
+	 * screen).
+	 *
+     * @since 7.5.0 Updated to support pagination of the related orders list.
+     *
+	 * @param WC_Subscription $subscription        The subscription whose related orders we are interested in.
+	 * @param int[]|null      $subscription_orders IDs of the related orders.
+	 * @param int|null        $page                The current page number.
+	 * @param int|null        $max_num_pages       The maximum number of pages in the set.
+	 *
+	 * @return void
+	 */
+	public static function get_related_orders_pagination_template( WC_Subscription $subscription, ?array $subscription_orders = null, ?int $page = null, ?int $max_num_pages = null ) {
```

Comment:
Double-checking my understanding here. I see that $subscription and $subscription_orders are not used in pagination template. Is it here for other extensions in case they want to have more advanced pagination? If that's the case, sure let's keep it, otherwise, we can limit pagination template to only two params used page and max_num_pages.

Reply:
The template doc block was incorrect, and implied that `$subscription` and `$subscription_orders` were in scope. That was wrong, so I've updated it:

https://github.com/Automattic/woocommerce-subscriptions-core/pull/833/commits/266e9b34d1d623a0fcff10029a6b3f929f953f99

Does that clarify things?